common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCacheWithMemoryCache.java

```diff
@@ -332,6 +339,10 @@ public boolean test(String value) {
                 // Remove the entry and return null...
                 Logger.warn(methodTag, "The returned Credential was uninitialized. Removing...");
                 mSharedPreferencesFileManager.remove(cacheKey);
+            } else if (AccessTokenRecord.class == clazz
+                    && !mSha1Cleared
```

Comment:
>mSha1Cleared [](http://example.com/codeflow?start=24&length=12)

Put this condition first since a bool check will be faster (and more often false) than the class equality check.